### PR TITLE
Clean up test run scripts

### PR DESCRIPTION
**Pull request by BruceForstall**
Remove unused, obsolete arguments.

Remove never-used jitdisasm support.

Add better support for finding Python in run.cmd (taken from build-runtime.cmd).

### CODE REVIEW

**Comment by BruceForstall** [comment]
I don't believe these scripts are used in the CI system anymore, just on local dev machines.

Many things have changed about the test build/run process, so things like the `test_native_bin_location` switch are no longer used.

File: src/tests/build.sh [hunk 1]
```diff
@@ -655,27 +655,19 @@ if [ "$__TargetOS" == "Android" ]; then
...
     echo "Tests run successful."
 else
-    echo "To run all tests use 'run.sh' where:"
-    echo "    testRootDir      = $__TestBinDir"
-    echo "    coreClrBinDir    = $__BinDir"
-    echo "    coreFxBinDir     = $CORE_ROOT"
-    echo "    testNativeBinDir = $__testNativeBinDir"
-    echo " -------------------------------------------------- "
-    echo " Example run.sh command"
+    echo "To run all tests use:"
```
[the remark this answers is not in the document]
Comment (BruceForstall) [reply src/tests/build.sh:668]: I wouldn't be opposed to that. The functionality only exists in build.sh, not build.cmd. Not sure who uses it (if anyone).

File: src/tests/run.py [hunk 2]
```diff
@@ -376,7 +357,7 @@ def __create_bash_wrapper__(self):
     echo \"CORE_ROOT set to ${CORE_ROOT}\"
 fi
 
-""" % (self.unique_name, self.core_root)
+""" % (self.unique_name, self.args.core_root)
```
Comment (BruceForstall) [inline src/tests/run.py:360]: This fixes an actual bug, which indicates that nobody ever ran this script on Linux!

File: src/tests/run.sh [hunk 3]
```diff
@@ -303,35 +210,29 @@ do
...
+# Set environment variables affecting tests.
+# (These should be run.py arguments.)
 ################################################################################
 
 if ((disableEventLogging == 0)); then
     export COMPlus_EnableEventLog=1
 fi
 
-export COMPlus_gcServer="$serverGC"
+if ((serverGC != 0)); then
```
Comment (BruceForstall) [inline src/tests/run.sh:221]: This avoids setting COMPlus_gcServer in the default case (0), to avoid unnecessary "noise"

**Review by trylek — APPROVED** [review]
LGTM, thanks for the fixes and cleanups!

**Comment by briansull** [comment]
I am seeing  a new failure when trying to run tests locally on my machine:

```
C:\fxkit\runtime>src\tests\run.cmd x64 checked
C:\Program Files (x86)\Microsoft Visual Studio\Shared\Python37_64\python.exe "C:\fxkit\runtime\src\tests\..\..\src\tests\run.py" -arch x64 -build_type Checked
'C:\Program' is not recognized as an internal or external command, operable program or batch file.
```

**Pull request merged**